src/project/mod.rs

```diff
@@ -624,6 +625,42 @@ impl Project {
         Ok(())
     }
 
+    /// Removes a dependency from `pixi.toml` based on `SpecType`.
+    pub fn remove_dependency(
+        &mut self,
+        dep: &PackageName,
+        spec_type: &SpecType,
+    ) -> miette::Result<(String, NamelessMatchSpec)> {
+        if let Item::Table(ref mut t) = self.doc[spec_type.name()] {
+            if t.contains_key(dep.as_normalized()) && t.remove(dep.as_normalized()).is_some() {
+                self.save()?;
```

Comment:
I would prefer that this method doesn't implicitly save the project to disk. I'd rather this is pulled out and done manually.